api_server.py

```
@@ -0,0 +1,1193 @@
+        success, result, status_code = perform_search(query, search_type, limit)
+        
+        if success:
+            return jsonify(result), status_code
+        else:
+            return jsonify(result), status_code
+            
+    except Exception as e:
+        logger.error(f"Search API endpoint failed: {str(e)}")
+        return jsonify({"error": f"Search failed: {str(e)}"}), 500
+        
+    except Exception as e:
```

Ruff (0.11.9)

try-except block with duplicate exception `Exception`

(B025)